ls1intum/Apollon
# Feature/create assessment drop zones

### Checklist

### Motivation and Context
Enables Artemis to drop SGIs on Apollon UML elements

### Description
Added drag events listener on assessable.tsx which listen for drag events. Only enabled when in assessment mode. If drag event is activated, it highlights the uml-elment in Apollon default select visualization. On drop: Try to parse the content of the 'text' MimeType, if it was added to the drag event into a Assessment object. If successful -> updates the assessment

### Steps for Testing


In Apollon:
1. `yarn prepare`
2. `cp package.json ./lib/`

Local installation of Apollon in Artemis (In Artemis):
1.package.json:
change `"@ls1intum/apollon": "2.0.x"`
to `"@ls1intum/apollon": "file:/your_path_to_prev_built_apollon_lib"`
2. Artemis: `yarn install`

Ticket Content:
1. Create an Modelling exercise in Artemis with SGI.
2. Drag SGI over UML-elements -> elements are selected
3. Dop -> Drop Target element is shown as assessed
4. DoubleClick on element -> popup opens with assessment (score and feedback of SGI)

### Screenshots
<img src="https://user-images.githubusercontent.com/22955066/78503312-276c4d00-7766-11ea-8f31-ff84e89b1b09.gif">


Comment @all: In Artemis, it's hard to see which element is selected as target for SGI drop. Either we could provide another representation of SGI on drag or highlight it differently. What are your opinions?
